SSPSolution/SSPSolution/LevelState.cpp

```diff
@@ -1139,21 +1226,26 @@ int LevelState::CreateLevel(LevelData::Level * data)
 
 		if (t_pc->PC_is_Static) {
 			StaticEntity* tse = new StaticEntity();
-			tse->SetGraphicsComponent(t_gc);
-			tse->SetPhysicsComponent(t_pc);
-			tse->Initialize(t_pc->PC_entityID, t_pc, t_gc);
+			//tse->SetGraphicsComponent(t_gc);
+			//tse->SetPhysicsComponent(t_pc);
+			//tse->SetAnimationComponent(t_anim);
+			tse->Initialize(t_pc->PC_entityID, t_pc, t_gc, t_anim);// Entity needs its ID
+
 			this->m_staticEntitys.push_back(tse); //Push new entity to list
 		}
 		else {
 			
 			DynamicEntity* tde = new DynamicEntity();
-			tde->SetGraphicsComponent(t_gc);
-			tde->SetPhysicsComponent(t_pc);
-			tde->Initialize(t_pc->PC_entityID, t_pc, t_gc);
+			//tde->SetGraphicsComponent(t_gc);
+			//tde->SetPhysicsComponent(t_pc);
+			//tde->SetAnimationComponent(t_anim);
+			tde->Initialize(t_pc->PC_entityID, t_pc, t_gc, t_anim);// Entity needs its ID
```

Comment:
Entities need to be initialised. This was a commit I did couple days ago. Recommitted as a fix.